mscore/instrwidget.cpp

```diff
@@ -479,6 +481,137 @@ void InstrumentsWidget::genPartList(Score* cs)
+                        // get the currently selected staff type of each combo and re-insert
+                        int numOfStaffListItems = item1->childCount();
+#if (!defined (_MSCVER) && !defined (_MSC_VER))
+                        int staffIdx[numOfStaffListItems];
+#else
+                        // MSVC does not support VLA. Replace with std::vector. If profiling determines that the
+                        //    heap allocation is slow, an optimization might be used.
+                        std::vector<int> staffIdx(numOfStaffListItems);
+#endif
+                        int itemIdx;
+                        for (itemIdx = 0; itemIdx < numOfStaffListItems; ++itemIdx)
+                              staffIdx[itemIdx] = (static_cast<StaffListItem*>(item1->child(itemIdx)))->staffTypeIdx();
+                        // do not consider hidden ones
+                        int plusIdx = delta;
+                        QTreeWidgetItem* nextParent = partiturList->topLevelItem(idx + plusIdx);
+                        while (nextParent && nextParent->isHidden()) {
+                              plusIdx += delta;
+                              nextParent = partiturList->topLevelItem(idx + plusIdx);
+                              }
+                        partiturList->insertTopLevelItem(idx + plusIdx, item1);
+                        // after-re-insertion, recreate each combo and set its index
+                        for (itemIdx=0; itemIdx < numOfStaffListItems; ++itemIdx) {
+                              StaffListItem* staffItem = static_cast<StaffListItem*>(item1->child(itemIdx));
+                              staffItem->initStaffTypeCombo(true);
+                              staffItem->setStaffType(staffIdx[itemIdx]);
+                              }
+                        partiturList->setItemExpanded(item1, isExpanded);
+                        }
+
+                  }
+            }
+      else { // CHILDREN_SELECTED
+            QTreeWidgetItem* parent = itemsToMove.front()->parent();
+            int idx = delta > 0 ? parent->childCount() - 1 : 0;
+            for ( ; (item = parent->child(idx)); idx += -delta) {
+                  if (!itemsToMove.contains(item))
+                        continue;
+                  
+                  int idx = parent->indexOfChild(item);
+                  int n = parent->childCount();
+                  // if staff item is not last of its part, move one slot down in part
+                  if ((idx + delta < n) && (idx + delta >= 0)) {
+                        StaffListItem* item1 = static_cast<StaffListItem*>(parent->takeChild(idx));
+                        // Qt looses the QComboBox set into StaffListItem when it is re-inserted into the tree:
+                        // get currently selected staff type and re-insert
+                        int staffTypeIdx = item1->staffTypeIdx();
+                        parent->insertChild(idx+delta, item1);
```

Comment:
space around operand

Reply:
these and the other 'space around operands' were already there. Fixed all that you commented on.

Reply:
No reason not to improve what you touch, but okay.

Reply:
You're right, I tried to be not to pedantic for my first contribution..